### PR TITLE
Improve Find/Replace shortcuts

### LINKED ISSUE #9142: Make Find/Replace easier to use
### Check for existing issues

- [X] Completed

### Describe the feature

In many cases, finding and replacing text in Zed requires more keystrokes than is really necessary, and sometimes confuses the user:
- If I want to replace something in a buffer, I have to hit `cmd-f` and then add `cmd-shift-h`, which feels like to many buttons for such a common task.
- If I want to find something in a buffer, I hit `cmd-f` and the Find panel appears so I can perform searching. Unless I replaced something recently, in which case the panel also contains the Replace line, and I have to additionally hit `cmd-shift-h` to close it. I know I can leave it open, but it confuses me, I don't want to replace something accidentally.

The same with the project search/replace actions.

Ohter editors allow the user to open find/replace panels in the exact form the user wants for the particular task with a single shortcut.

The proposal is to:

**Find/Replace in buffer**

- `cmd-f`:
  - In an editor opens the F/R panel without the replacement line.
  - In the F/R panel moves focus to the Search field if it's not already there.
- `cmd-alt-f`
  - In an editor opens the F/R panel with the replacement line.
  - In the F/R panel: if the replace line is hidden, shows it and moves focus to the Replace field, otherwies hides the replace line and moves focus to the Search field.

**Find/Replace in project**

Absolutely the same behaviour, but the default keybindings are `cmd-shift-f` and `cmd-shift-h`.

**What we break**

The `cmd-f...cmd-shift-h` sequence for "replace in buffer" will not work anymore. It'll be replaced with `cmd-alt-f`.

**What we get**

- the same approach to find/replace as in the most popular IDEs/editors.
- keybindings as in VSCode and Sublime Text.

**Alternative keybindings**

To tell the truth, I'd prefer the JB's keybindings for the described actions (`cmd-f` / `cmd-r` for buffer and `cmd-shift-f` / `cmd-shift-r` for project), but `cmd-r` is already reserved in Zed for the right dock toggling.

### If applicable, add mockups / screenshots to help present your vision of the feature

N/A

### PR DESCRIPTION
**Pull request by alygin**
This PR changes ways the Find/Replace functionality in the Buffer/Project Search is accessible via shortcuts. It makes those panels work the same way as in VS Code and Sublime Text.

The details are described in the issue: [Make Find/Replace easier to use](https://github.com/zed-industries/zed/issues/9142)

There's a difficulty with the Linux keybindings:

VS Code uses on MacOS (this PR replicates it):

| Action | Buffer Search | Project Search |
| --- | --- | --- |
| Find | `cmd-f` | `cmd-shift-f` |
| Replace | `cmd-alt-f` | `cmd-shift-h` |

VS Code uses on Linux (this PR replicates all but one):

| Action | Buffer Search | Project Search |
| --- | --- | --- |
| Find | `ctrl-f` | `ctrl-shift-f` |
| Replace | `ctrl-h` ❗ | `ctrl-shift-h` |

The problem is that `ctrl-h` is already taken by the `editor::Backspace` action in Zed on Linux.

There's two options here:

1. Change keybinding for `editor::Backspace` on Linux to something else, and use `ctrl-h` for the "replace in buffer" action.
2. Use some other keybinding on Linux in Zed. This PR introduces `ctrl-r` for this purpose, though I'm not sure it's the best choice. 

What do you think?

fixes #9142

Release Notes:

- Improved access to "Find/Replace in Buffer" and "Find/Replace in Files" via shortcuts (#9142).

Optionally, include screenshots / media showcasing your addition that can be included in the release notes.

- N/A

### CODE REVIEW

File: assets/keymaps/default-linux.json [hunk 1]
```diff
@@ -212,7 +213,9 @@
       "enter": "search::SelectNextMatch",
       "shift-enter": "search::SelectPrevMatch",
       "alt-enter": "search::SelectAllMatches",
-      "alt-tab": "search::CycleMode"
+      "alt-tab": "search::CycleMode",
+      "ctrl-f": "search::FocusSearch",
+      "ctrl-r": "search::ToggleReplace"
```
Comment (mikayla-maki) [inline assets/keymaps/default-linux.json:218]: Let's change this to ctrl-h

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

**Comment by alygin** [comment]
> We should remove the `ctrl-h` binding to backspace on Linux.

What about `ctrl-h` binding to `zed::Hide` on Linux? This is a binding in the global context, so it has no direct conflict with Find/Replace in buffer, yet it fires when there's no open editor. For instance, when the project panel is activated and the user hits `ctrl-h`. It's quite confusing.

I see the _"// todo(linux): Review the editor bindings"_ comment, so these bindings are probably not set in stone. I found that Ubuntu uses `alt-f9` to minimize the current window, but I'm not sure about other distros and if there's a consensus on this in Linux.

I'll comment this binding. If someone has an idea on what it should be, let's change it.

File: assets/keymaps/default-linux.json (hunk 1 above)
Comment (alygin) [reply assets/keymaps/default-linux.json]: Done.

**Comment by mikayla-maki** [comment]
I say just change it to alt-f9 and let’s fix forward if someone files an issue :D

**Comment by alygin** [comment]
> I say just change it to alt-f9 and let’s fix forward if someone files an issue :D

OK :)
Done

**Review by mikayla-maki — APPROVED** [review]

**Pull request merged**